cli/src/main/scala-2.12/coursier/cli/Helper.scala

```diff
@@ -291,10 +291,27 @@ class Helper(
     .flatMap {
       case (dep, extraParams) =>
         extraParams.get("url").map { url =>
-          dep.moduleVersion -> (new URL(URLDecoder.decode(url, "UTF-8")), true)
+          dep.moduleVersion -> (convertStringToUrl(url), true)
         }
     }.toMap
 
+  private def convertStringToUrl(url: String): URL = {
+    if (url.startsWith("local"))
+      new URL(null, URLDecoder.decode(url, "UTF-8"), new URLStreamHandler {
+        override def openConnection(u: URL): URLConnection = {
+          val toFileUrl = new URL(url.toString.replace("local", "file"))
+          toFileUrl.openConnection()
+        }
+      })
+    else
+      new URL(URLDecoder.decode(url, "UTF-8"))
+  }
+
+  depsWithUrls.foreach { case (moduleVersion: (Module, String), urlBool: (URL, Boolean)) =>
+    if (!cacheFileArtifacts && urlBool._1.toString.startsWith("file:/"))
+      errPrintln(s"Warning: Please use the 'local:/' protocol for ${urlBool._1.toString}")
```

Comment:
Example error msg I saw:
```
Warning: Please use the 'local:/' protocol for file:/Users/yic/.m2/repository/junit/junit/4.12/junit-4.12.jar
```
nit: 
* `local:/` -> `local://`
* `file:` -> `file://`
* to make the message a bit more clear, append `to prevent it from being copied into cache.`

Reply:
Same as above, `local:/` and `file:/` are ok.